storm-core/src/clj/backtype/storm/command/blobstore.clj

```diff
@@ -0,0 +1,163 @@
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+(ns backtype.storm.command.blobstore
+  (:import [java.io InputStream OutputStream])
+  (:use [backtype.storm config])
+  (:import [backtype.storm.generated SettableBlobMeta AccessControl AuthorizationException
+            KeyNotFoundException])
+  (:import [backtype.storm.blobstore BlobStoreAclHandler])
+  (:use [clojure.string :only [split]])
+  (:use [clojure.tools.cli :only [cli]])
+  (:use [clojure.java.io :only [copy input-stream output-stream]])
+  (:use [backtype.storm blobstore log util])
+  (:gen-class))
+
+(defn update-blob-from-stream
+  "Update a blob in the blob store from an InputStream"
+  [key ^InputStream in]
+  (with-configured-blob-client blobstore
+    (let [out (.updateBlob blobstore key)]
+      (try 
+        (copy in out)
+        (.close out)
+        (catch Exception e
+          (log-message e)
+          (.cancel out)
+          (throw e))))))
+
+(defn create-blob-from-stream
+  "Create a blob in the blob store from an InputStream"
+  [key ^InputStream in ^SettableBlobMeta meta]
+  (with-configured-blob-client blobstore
+    (let [out (.createBlob blobstore key meta)]
+      (try 
+        (copy in out)
+        (.close out)
+        (catch Exception e
+          (.cancel out)
+          (throw e))))))
+
+(defn read-blob
+  "Read a blob in the blob store and write to an OutputStream"
+  [key ^OutputStream out]
+  (with-configured-blob-client blobstore
+    (with-open [in (.getBlob blobstore key)]
+      (copy in out))))
+
+(defn as-access-control
+  "Convert a parameter to an AccessControl object"
+  [param]
+  (BlobStoreAclHandler/parseAccessControl (str param)))
+
+(defn as-acl
+  [param]
+  (map as-access-control (split param #",")))
+
+(defn access-control-str
+  [^AccessControl acl]
+  (BlobStoreAclHandler/accessControlToString acl))
+
+(defn read-cli [args]
+  (let [[{file :file} [key] _] (cli args ["-f" "--file" :default nil])]
+    (if file
+      (with-open [f (output-stream file)]
+        (read-blob key f))
+      (read-blob key System/out))))
+
+(defn update-cli [args]
+  (let [[{file :file} [key] _] (cli args ["-f" "--file" :default nil])]
+    (if file
+      (with-open [f (input-stream file)]
+        (update-blob-from-stream key f))
+      (update-blob-from-stream key System/in))
+    (log-message "Successfully updated " key)))
+
+(defn create-cli [args]
+  (let [[{file :file acl :acl replication-factor :replication-fctr} [key] _] (cli args ["-f" "--file" :default nil]
+                                                  ["-a" "--acl" :default [] :parse-fn as-acl]
+                                                  ["-r" "--replication-factor" :default -1 :parse-fn parse-int])
```

Comment:
* Check white-space/indentation/format.  This looks odd.
* Is `:replication-fctr` the correct key name, or has that been changed?